### DISCUSSION TITLE
Switching between 2 specific beatmaps don't switch playing track

### DISCUSSION THREAD

**Discussion by Feodor0090** [ppy/osu#19667] [Q&A] [answered]
TL;DR: switching between 2 specific beatmaps with **different** audio tracks don't switch them, it's also possible to play one map with music from another.

# Steps

1. Have fresh osu!lazer `2022.731.1` install.
2. Download any beatmap (%0)
3. Download https://osu.ppy.sh/beatmapsets/57762 (%1), then https://osu.ppy.sh/beatmapsets/228109 (%2).
4. Set sorting by date added.
5. Select %0, then %1, then %0, then %2. Note that all three beatmaps have different tracks.
6. Select %0, then %1, then %2. Note that playing track did not change, %1 is playing.
7. Select %0, then %2, then %1. Track did not change, %2 is still playing.
8. Open Now Playing overlay. Note that it correctly shows %1. Try to seek - you will still hear %2.
9. Select %0. Open Now Playing and click next track button twice. The same, %1 is playing, %2 is shown, seeking is done on %1.
10. Select %0, then %1, then %2. Press ctrl+enter. %2 will be played with music from %1. It also works in opposite order.
11. Select %0, %1, %2. Try to edit selected one. Editor will open %2 with music from %1, again.

# Log 

[runtime.log](https://github.com/ppy/osu/files/9289987/runtime.log)

# Video
Without step 11, logs were lost (provided one is doing described steps again). BMs are downloaded before.

https://user-images.githubusercontent.com/53872073/183639173-dbb80db0-36ef-4da0-baae-62c1fd2a9491.mp4

# DB
[client.realm.zip](https://github.com/ppy/osu/files/9290019/client.realm.zip)

**Comment by peppy** [comment, marked as answer]
Thanks for the comprehensive reproduction instructions. I have reproduced and found the issue.

**Reply by peppy** [reply]
Will be fixed via #19678.
